### PR TITLE
Disable small sector erase size for UBIFS on flash

### PR DESCRIPTION
**Pull request by bmridul**
UBIFS does not work on NOR Flash with small sector erasesize. 
https://cateee.net/lkddb/web-lkddb/MTD_SPI_NOR_USE_4K_SECTORS.html

The kernel config CONFIG_MTD_SPI_NOR_USE_4K_SECTORS, which is enabled by default is removed.

Unit Test:

    cisco@sonic:~$ grep CONFIG_MTD_SPI_NOR_USE_4K_SECTORS /boot/config-5.10.0-18-2-amd64
    # CONFIG_MTD_SPI_NOR_USE_4K_SECTORS is not set
    cisco@sonic:~$ findmnt -n /mnt/obfl
    /mnt/obfl ubi0   ubifs  rw,relatime,assert=read-only,ubi=0,vol=0

### CODE REVIEW

**Comment by bmridul** [comment]
@abdosi,  Pls check.

**Review by paulmenzel — COMMENTED** [review]
Thank you for your patch. I’d be great if you could amend the git commit message summary and merge/pull request title to use imperative mood and *to enable* or *for*:

> Disable small sector erase size for UBIFS on flash

Also could you please copy the merge/pull request description to the commit message?

I’d use Markdown to format the description, and, for example, indent code blocks by four spaces.

    # mount | grep obfl
    ubi0 on /mnt/obfl type ubifs (rw,relatime,assert=read-only,ubi=0,vol=0)

And just a tip, that `findmnt -n /mnt/obfl` does the same. (`-n` suppresses the printing of the header.)

**Comment by bmridul** [comment]
> Thank you for your patch. I’d be great if you could amend the git commit message summary and merge/pull request title to use imperative mood and _to enable_ or _for_:
> 
> > Disable small sector erase size for UBIFS on flash
> 
> Also could you please copy the merge/pull request description to the commit message?
> 
> I’d use Markdown to format the description, and, for example, indent code blocks by four spaces.
> 
> ```
> # mount | grep obfl
> ubi0 on /mnt/obfl type ubifs (rw,relatime,assert=read-only,ubi=0,vol=0)
> ```
> 
> And just a tip, that `findmnt -n /mnt/obfl` does the same. (`-n` suppresses the printing of the header.)

@paulmenzel , Pls check updates

**Review by abdosi — DISMISSED** [review]

**Comment by prgeor** [comment]
@bmridul Why a filesystem like UBIFS is limited by Nand's/NOR's erase size? What error are you seeing with this config enbled?

**Comment by prgeor** [comment]
@bmridul any reference for this claim:
`UBIFS does not work on NOR Flash with small sector erasesize`

**Comment by prgeor** [comment]
@bmridul why should UBIFS care the underlying media NOR or NAND

**Comment by bmridul** [comment]
> @bmridul any reference for this claim: `UBIFS does not work on NOR Flash with small sector erasesize`

@prgeor , I have put a [link](https://cateee.net/lkddb/web-lkddb/MTD_SPI_NOR_USE_4K_SECTORS.html) to the man  page for the kernel config. There itself this issue is mentioned. 

    Please note that some tools/drivers/filesystems may not work with 4096 B erase size (e.g. UBIFS requires 15 KiB as a minimum).

**Comment by bmridul** [comment]
> @bmridul Why a filesystem like UBIFS is limited by Nand's/NOR's erase size? What error are you seeing with this config enbled?

Basically the ubiattach command fails with errors..

    **[  145.820167] ubi0 error: validate_vid_hdr [ubi]: bad data_size
    [  145.826621] ubi0 error: validate_vid_hdr [ubi]: bad VID header**
    [  145.833163] Volume identifier header dump:
    [  145.837752]  magic     55424921
    [  145.841270]  version   1
    [  145.844109]  vol_type  1
    [  145.846934]  copy_flag 1
    [  145.849775]  compat    5
    [  145.852609]  vol_id    2147479551
    [  145.856318]  lnum      0
    [  145.859146]  data_size 22016
    [  145.862370]  used_ebs  0
    [  145.865207]  data_pad  0
    [  145.868044]  sqnum     1
    [  145.870869]  hdr_crc   11f8b00c
    [  145.874390] Volume identifier header hexdump:
    **[  145.879362] ubi0 error: ubi_io_read_vid_hdr [ubi]: validation failed for PEB 2064**
    [  145.891655] ubi0 error: ubi_attach_mtd_dev [ubi]: failed to attach mtd4, error -22

With the fix, we are able to mount the flash and access the file system contents.

File: patch/kconfig-exclusions [hunk 1]
```diff
@@ -13,6 +13,8 @@ CONFIG_CGROUP_NET_PRIO
 # Unset X86_PAT according to Broadcom's requirement
 CONFIG_X86_PAT
 CONFIG_MLXSW_PCI
+# Unset to enable UBIFS on flash
+CONFIG_MTD_SPI_NOR_USE_4K_SECTORS
```
Comment (saiarcot895) [inline patch/kconfig-exclusions:17]: Can you move this to the common section, since this issue can affect non-amd64 architectures as well?

Comment (bmridul) [reply patch/kconfig-exclusions:17]: Ack.

**Comment by paulmenzel** [comment]
Why not just cite the line from the Kconfig description?

> From the description of `CONFIG_MTD_SPI_NOR_USE_4K_SECTORS`:
> 
> > Please note that some tools/drivers/filesystems may not work with 4096 B erase size (e.g. UBIFS requires 15 KiB as a minimum).

Also, I’d appreciate the UBIFS “error” messages in the commit message too, so should somebody hit this error, it is easily found in the git history/log.

**Review by prgeor — APPROVED** [review]

**Comment by abdosi** [comment]
@paulmenzel / @saiarcot895 can we help merge of this

**Comment by paulmenzel** [comment]
Thank you.

**Pull request merged**